pkg/gcp/gcp.go

```diff
@@ -89,24 +102,40 @@ func newOptions() *options {
 
 // NewClient creates a new GCP storage client. The Client will automatically look for the Google Application
 // Credential environment variable or look for the Google Application Credential file.
-func NewClient(ctx context.Context, opts ...Option) (*GCSClient, error) {
+func NewClient(ctx context.Context, bucket *sourcev1.Bucket, opts ...Option) (*GCSClient, error) {
 	o := newOptions()
 	for _, opt := range opts {
 		opt(o)
 	}
 
 	var clientOpts []option.ClientOption
 
+	// Authentication priority: Secret → AuthOpts (Workload Identity)
 	switch {
 	case o.secret != nil && o.proxyURL == nil:
 		clientOpts = append(clientOpts, option.WithCredentialsJSON(o.secret.Data["serviceaccount"]))
-	case o.proxyURL != nil:
+	case o.secret != nil && o.proxyURL != nil:
 		httpClient, err := o.newCustomHTTPClient(ctx, o)
 		if err != nil {
 			return nil, err
 		}
 		clientOpts = append(clientOpts, option.WithHTTPClient(httpClient))
+	case len(o.authOpts) > 0:
```

Comment:
```suggestion
	default:
```

If `.spec.provider` is `gcp` and `.spec.secretRef` is not set, we should use the `auth` package logic